HLTrigger/special/src/HLTMultipletFilter.cc

```diff
@@ -0,0 +1,177 @@
+  minPt_             = iConfig.getParameter<double> ("MinPt"); 
+
+  if (hltEGammaSeedLabel_ == edm::InputTag()) {
+    flag_[EGamma]   = false;
+  } else {
+    flag_[EGamma]   = true;
+    hltEGammaToken_ = consumes<l1t::EGammaBxCollection>(hltEGammaSeedLabel_);
+  }
+  if (hltEtSumSeedLabel_ == edm::InputTag()) {
+    flag_[EtSum]    = false;
+  } else {
+    flag_[EtSum]    = true;
+    hltEtSumToken_  = consumes<l1t::EtSumBxCollection>(hltEtSumSeedLabel_);
+  }
+  if (hltJetSeedLabel_ == edm::InputTag()) {
+    flag_[Jet]      = false;
+  } else {
+    flag_[Jet]      = true;
+    hltJetToken_    = consumes<l1t::JetBxCollection>(hltJetSeedLabel_);
+  }
+  if (hltMuonSeedLabel_ == edm::InputTag()) {
+    flag_[Muon]     = false;
+  } else {
+    flag_[Muon]     = true;
+    hltMuonToken_   = consumes<l1t::MuonBxCollection>(hltMuonSeedLabel_);
+  }
+  if (hltTauSeedLabel_ == edm::InputTag()) {
+    flag_[Tau]      = false;
+  } else {
+    flag_[Tau]      = true;
+    hltTauToken_ = consumes<l1t::TauBxCollection>(hltTauSeedLabel_);
+  }
+  edm::LogVerbatim("Report") << "Input Parameters:: minN = " << minN_
+			     << " minPt = " << minPt_ << " Eta " << minEta_
+			     << ":" << maxEta_ << " Phi " << minPhi_ << ":" 
+			     << maxPhi_ << " GT Seed for EGamma " 
+			     << hltEGammaSeedLabel_ << ", EtSum "
+			     << hltEtSumSeedLabel_ << ", Jet "
+			     << hltJetSeedLabel_ << ", Muon "
+			     << hltMuonSeedLabel_ << ", and Tau "
+			     << hltTauSeedLabel_ << std::endl;
+}
+
+HLTMultipletFilter::~HLTMultipletFilter()= default;
+
+void
+HLTMultipletFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("L1EGammaInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1EtSumInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1JetInputTag",edm::InputTag("hltCaloStage2Digis:Jet"));
+  desc.add<edm::InputTag>("L1MuonInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1TauInputTag",edm::InputTag("hltCaloStage2Digis:Tau"));
+  desc.add<int>("MinN",1);
+  desc.add<double>("MinEta",1.305);
+  desc.add<double>("MaxEta",3.000);
+  desc.add<double>("MinPhi",5.4105);
+  desc.add<double>("MaxPhi",5.5796);
+  desc.add<double>("MinPt", 20.0);
+  descriptions.add("hltMultipletFilter",desc);
+}
+
+bool HLTMultipletFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSetup, trigger::TriggerFilterObjectWithRefs & filterproduct) const {
+
+  // the filter object
+  if (saveTags()) {
+    for (int type=0; type<nobj_; ++type) {
+      if (flag_[type]) {
+	switch (type) {
+	  case EGamma:
+	    filterproduct.addCollectionTag(hltEGammaSeedLabel_);
+	    break;
+	  case EtSum:
+	    filterproduct.addCollectionTag(hltEtSumSeedLabel_);
+	    break;
+	  case Jet:
+	    filterproduct.addCollectionTag(hltJetSeedLabel_);
+	    break;
+ 	  case Muon:
+	    filterproduct.addCollectionTag(hltMuonSeedLabel_);
+	    break;
+	  default:
```

Comment:
Why default?

And why not just 5 if statements, on flag_[0] to flag_[4], without any loop and any case statement?